### PR TITLE
ERR: Raise on invalid na_action in Series.map

### PR DESCRIPTION
**Pull request by dsaxton**
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry

`Series.map` treats any `na_action` other than `"ignore"` like `None`, but it should probably raise if someone provides an invalid option:

```python
In [1]: import pandas as pd                                                                                                                                

In [2]: pd.Series([1, 2, 3]).map(lambda x: x, na_action="xxxxx")                                                                                           
Out[2]: 
0    1
1    2
2    3
dtype: int64
```

### CODE REVIEW

**Comment by jbrockmendel** [comment]
Do any Index methods need the same fix?

**Comment by dsaxton** [comment]
> Do any Index methods need the same fix?

Looks like they do. Actually it seems worse, e.g. `ExtensionArray` indexes just ignore `na_action`: https://github.com/pandas-dev/pandas/blob/master/pandas/core/indexes/extension.py#L291

File: pandas/core/base.py [hunk 1]
```diff
@@ -1171,8 +1171,10 @@ def _map_values(self, mapper, na_action=None):
                 def map_f(values, f):
                     return lib.map_infer_mask(values, f, isna(values).view(np.uint8))
 
-            else:
+            elif na_action is None:
                 map_f = lib.map_infer
+            else:
+                raise ValueError("na_action must either be 'ignore' or None")
```
Comment (jreback) [inline pandas/core/base.py]: add: ```f`{na_action}``` was passed

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/series/test_apply.py [hunk 2]
```diff
@@ -787,3 +787,9 @@ def test_map_float_to_string_precision(self):
         result = ser.map(lambda val: str(val)).to_dict()
         expected = {0: "0.3333333333333333"}
         assert result == expected
+
+    def test_map_with_invalid_na_action_raises(self):
+        s = pd.Series([1, 2, 3])
```
Comment (jreback) [inline pandas/tests/series/test_apply.py]: add the issue umber as a comment

**Review by jreback — APPROVED** [review]
lgtm. merge master and ping on green.

**Comment by dsaxton** [comment]
> lgtm. merge master and ping on green.

@jreback Merged master and green, thanks for reviewing

**Comment by jreback** [comment]
thanks

**Pull request merged**